starboard/nplb/posix_compliance/posix_time_clock_test.cc

```diff
@@ -0,0 +1,181 @@
+// Copyright 2025 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <time.h>
+#include <unistd.h>
+
+#include <atomic>
+#include <type_traits>
+
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace starboard {
+namespace nplb {
+namespace {
+
+const int64_t kMicrosecondsPerSecond = 1'000'000LL;
+
+// Work time long enough to ensure measurable CPU time usage.
+constexpr long kMinimumWorkTimeMicroseconds = 100'000;  // 100ms
+
+static int64_t TimespecToMicroseconds(const struct timespec& ts) {
+  return (static_cast<int64_t>(ts.tv_sec) * kMicrosecondsPerSecond) +
+         (static_cast<int64_t>(ts.tv_nsec) / 1'000LL);
+}
+
+// Helper function to do CPU work for until at least work_time_microseconds
+// elapsed on the monotonic wall clock. Return the elapsed time in microseconds.
+// Note: This function uses CLOCK_MONOTONIC to measure the elapsed
+// time, not the CPU time clock. This is intentional, as the goal is to
+// consume CPU time for a specific duration as measured by wall clock time.
+// The CPU time clock is then checked to see how much CPU time was actually
+// consumed.
+long ConsumeCpuForDuration(long work_time_microseconds) {
+  struct timespec ts_before = {};
```

Comment:
Is there a reason why we do `struct timespec foo = {}` here vs `struct timespec foo {}` above?